### PR TITLE
Feature/send raw log

### PR DESCRIPTION
**Pull request by itaymmguardicore**
# Feature
> Changes/Fixes the following feature
You can now download the monkey's raw log from the Monkey Island.

* [X] Have you added an explanation of what your changes do and why you'd like to include them?
* [X] Have you successfully tested your changes locally?

##  Changes
  - Monkeys send their log to the island.
  - Users can download the log from the map's preview panel.

### CODE REVIEW

**Comment by danielguardicore** [comment]
General comments:

- Why are we only uploading the log file at end, rather than the ability to request monkey logs? If the monkey bugs out, we won't have the logs.
- Why are we B64ing the log? The log itself is text only, we can just send it over as plain text (unicode?)

**Comment by itaymmguardicore** [comment]
@danielguardicore All fixed

File: monkey_island/cc/ui/src/components/map/preview-pane/PreviewPane.js [hunk 1]
```diff
@@ -82,22 +82,34 @@ class PreviewPaneComponent extends React.Component {
...
+                  onChange={(e) => this.forceKill(e, asset)}/>
 
         </td>
       </tr>
     );
   }
 
-  downloadLog(asset) {
+  unescapeLog(st) {
+    return st.substr(1, st.length - 2) // remove quotation marks on beginning and end of string.
```
Comment (danielguardicore) [inline monkey_island/cc/ui/src/components/map/preview-pane/PreviewPane.js]: Isn't there a library for that? or JSON unparse?

File: chaos_monkey/main.py [hunk 2]
```diff
@@ -78,19 +79,19 @@ def main():
...
-                WormConfiguration.dropper_log_path_windows) if sys.platform == "win32" else WormConfiguration.dropper_log_path_linux
+            log_path = utils.get_dropper_log_path()
             monkey_cls = MonkeyDrops
         else:
             return True
     except ValueError:
         return True
 
     if WormConfiguration.use_file_logging:
+        if os.path.exists(log_path):
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply chaos_monkey/main.py]: You don't want to get the logs from a previous run (and you don't want the log file to get huge)

File: monkey_island/cc/services/log.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    def remove_logs_by_monkey_id(monkey_id):
+        fs = gridfs.GridFS(mongo.db)
+        for log in mongo.db.log.find({'monkey_id': monkey_id}):
+            fs.delete(log['file_id'])
+        mongo.db.log.delete_many({'monkey_id': monkey_id})
+
+    @staticmethod
+    def add_log(monkey_id, log_data, timestamp=datetime.now()):
+        LogService.remove_logs_by_monkey_id(monkey_id)
+        fs = gridfs.GridFS(mongo.db)
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply monkey_island/cc/services/log.py]: fixed

File: monkey_island/cc/resources/root.py [hunk 4]
```diff
@@ -33,7 +33,8 @@ def get_server_info():
 
     @staticmethod
     def reset_db():
-        [mongo.db[x].drop() for x in ['config', 'monkey', 'telemetry', 'node', 'edge', 'report']]
+        [mongo.db[x].drop() for x in
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply monkey_island/cc/resources/root.py]: fixed

File: monkey_island/cc/services/log.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                {
+                    'monkey_label': monkey_label,
+                    'log': log_file.read(),
+                    'timestamp': log['timestamp']
+                }
+
+    @staticmethod
+    def remove_logs_by_monkey_id(monkey_id):
+        fs = gridfs.GridFS(mongo.db)
+        for log in mongo.db.log.find({'monkey_id': monkey_id}):
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply monkey_island/cc/services/log.py]: fixed

**Comment by danielguardicore** [comment]
LGTM

**Pull request merged**